doc: add flux-resource(1),  clean up help output

garlick opened the pull request:
As discussed in #4000, `flux help` prints some commands that are probably not going to be very popular.  Drop those from the output, and add a reference to our [readthedocs documentation](https://flux-framework.readthedocs.io).  Also add a missing `flux-resource(1)` man page.  There are others but that one seemed egregious. 

After these changes, output is:
```
$ flux help
Usage: flux [OPTIONS] COMMAND ARGS
  -h, --help             Display this message.
  -v, --verbose          Be verbose about environment and command search
  -V, --version          Display command and component versions
  -p, --parent           Set environment of parent instead of current instance

Common commands from flux-core:
   cron               Schedule tasks on timers and events
   dmesg              manipulate broker log ring buffer
   env                Print the flux environment or execute a command inside it
   exec               Execute processes across flux ranks
   {get,set,ls}attr   Access, modify, and list broker attributes
   jobs               list jobs submitted to Flux
   resource           list/manipulate Flux resource status
   kvs                Flux key-value store utility
   mini               Minimal Job Submission Tool
   job                Job Housekeeping Tool
   ping               measure round-trip latency to Flux services
   proxy              Create proxy environment for Flux instance
   start              bootstrap a local Flux instance
   version            Display flux version information

See also: https://flux-framework.readthedocs.io
```

chu11 commented on doc/man1/flux-broker.rst:
```diff
@@ -1,5 +1,3 @@
-.. flux-help-description: Invoke Flux message broker daemon
```
commit message typo exected -> expected

chu11 reviewed (approved):
LGTM, only noticed this one nit.

grondo commented on src/cmd/cmdhelp.c:
```diff
@@ -231,6 +231,8 @@ void emit_command_help (const char *plist, FILE *fp)
 
     free (argz);
+
+    fprintf (fp, "\nSee also: https://flux-framework.readthedocs.io\n");
```
This will require the `'flux-help command list can be extended'` test in `t0001-basic.t` to be modified, since unfortunately the test compares verbatim output from `flux help`.

garlick commented:
Just pushed the following changes
- fixed commit typo identified by @chu11
- relocated readthedocs URL output from flux(1) to a better spot that also addresses the t0001-basic.t test failure
- added flux-resource(1) to the readthedocs index (oops)
- reorganized flux-resource(1) to hopefully more clearly distinguish between what guests may want vs sys admins

garlick commented:
Oh, and also just pushed the same change to requirements.txt that we pushed to flux-docs and flux-rfc, to fix bullet lists that weren't rendered properly.

grondo reviewed (approved):
LGTM! Thanks for the addition of `flux-resource(1)`!

garlick commented:
Thanks.  I guess I'll set MWP then.
